lib/models/blueprint.js

```diff
@@ -849,29 +849,80 @@ Blueprint.prototype._locals = function(options) {
   var moduleName = options.entity && options.entity.name || packageName;
   var sanitizedModuleName = moduleName.replace(/\//g, '-');
 
-  return new Promise(function(resolve) {
-    resolve(this.locals(options));
-  }.bind(this)).then(function (customLocals) {
-    var fileMapVariables = this._generateFileMapVariables(moduleName, customLocals, options);
-    var fileMap = this.generateFileMap(fileMapVariables);
-    var standardLocals = {
-      dasherizedPackageName: stringUtils.dasherize(packageName),
-      classifiedPackageName: stringUtils.classify(packageName),
-      dasherizedModuleName: stringUtils.dasherize(moduleName),
-      classifiedModuleName: stringUtils.classify(sanitizedModuleName),
-      camelizedModuleName: stringUtils.camelize(sanitizedModuleName),
-      decamelizedModuleName: stringUtils.decamelize(sanitizedModuleName),
-      fileMap: fileMap,
-      hasPathToken: this.hasPathToken,
-      targetFiles: options.targetFiles,
-      rawArgs: options.rawArgs
-    };
-
-    return merge({}, standardLocals, customLocals);
+  return this._processPrompts(options).then(function (prompts) {
+    return new Promise(function(resolve) {
+      resolve(this.locals(prompts));
+    }.bind(this)).then(function (customLocals) {
+      var fileMapVariables = this._generateFileMapVariables(moduleName, customLocals, options);
+      var fileMap = this.generateFileMap(fileMapVariables);
+      var standardLocals = {
+        dasherizedPackageName: stringUtils.dasherize(packageName),
+        classifiedPackageName: stringUtils.classify(packageName),
+        dasherizedModuleName: stringUtils.dasherize(moduleName),
+        classifiedModuleName: stringUtils.classify(sanitizedModuleName),
+        camelizedModuleName: stringUtils.camelize(sanitizedModuleName),
+        decamelizedModuleName: stringUtils.decamelize(sanitizedModuleName),
+        fileMap: fileMap,
+        hasPathToken: this.hasPathToken,
+        targetFiles: options.targetFiles,
+        rawArgs: options.rawArgs
+      };
+
+      return merge({}, standardLocals, customLocals);
+    }.bind(this));
   }.bind(this));
```

Comment:
why do you think that?

the `new Promise()` wrapping code was previously used to catch errors in `this.locals()`, but any code run in a `.then()` will automatically get caught and passed down the promise chain.

In my proposal I'm using the same trick to catch errors in your new `this._processPrompts()` method. Although since that method is supposed to be private I'm not sure it's even needed.